core/src/main/java/net/sourceforge/jnlp/tools/JarCertVerifier.java

```diff
@@ -415,6 +415,24 @@ VerifyResult verifyJarEntryCerts(final String jarPath, final boolean jarHasManif
         return result;
     }
 
+    private boolean isTrustedTsa(CertPath certPath) {
+        try {
+            final List<KeyStore> caKeyStores = KeyStores.getCAKeyStores();
+            // Check entire cert path for a trusted CA
+            for (final Certificate c : certPath.getCertificates()) {
+                if (c instanceof X509Certificate) {
+                    final X509Certificate x509 = (X509Certificate) c;
+                    if (CertificateUtils.inKeyStores(x509, caKeyStores)) {
+                        return true;
+                    }
+                }
+            }
+        } catch (RuntimeException e) {
+            LOG.warn("Unable to read through CA cert store files.");
```

Comment:
add e

Reply:
and maybe add certPath in log?

Reply:
removed try catch since there is no method thrown anything, not even a RuntimeException